tensor2tensor/trax/layers/initializers.py

```diff
@@ -30,21 +41,70 @@ def Init(shape, rng):
   return Init
 
 
-def GlorotNormalInitializer(out_dim=0, in_dim=1, scale=onp.sqrt(2)):
-  """An initializer function for random Glorot-scaled coefficients."""
+def RandomUniformInitializer(lim=1.0):
+  """An initializer function for random uniform coefficients."""
   def Init(shape, rng):
-    fan_in, fan_out = shape[in_dim], shape[out_dim]
-    size = onp.prod(onp.delete(shape, [in_dim, out_dim]))
-    std = scale / backend.numpy.sqrt((fan_in + fan_out) / 2. * size)
-    return (std * backend.random.normal(rng, shape)).astype('float32')
+    return (backend.random.uniform(rng, shape, -lim, lim)).astype('float32')
   return Init
 
 
-def GlorotUniformInitializer(out_dim=0, in_dim=1):
-  """An initializer function for random uniform Glorot-scaled coefficients."""
+def VarianceScalingInitializer(out_dim, in_dim, scale, mode, distribution):
+  """Initializer capable of adapting its scale to the shape of weights tensors."""
+  if scale <= 0.:
+    raise ValueError('scale must be positive float, {} given'.format(scale))
+  if mode not in {'fan_in', 'fan_out', 'fan_avg'}:
+    raise ValueError('Invalid mode argument:, {}, must be either fan_in, fan_out or fan_avg'.format(mode))
+
   def Init(shape, rng):
-    fan_in, fan_out = shape[in_dim], shape[out_dim]
-    std = backend.numpy.sqrt(2.0 / (fan_in + fan_out))
-    a = backend.numpy.sqrt(3.0) * std
-    return backend.random.uniform(rng, shape, minval=-a, maxval=a)
-  return Init
+    fan_in, fan_out = _get_fans(shape, out_dim, in_dim)
+    gain = scale
+    if mode == 'fan_in':
+      gain /= fan_in
+    elif mode == 'fan_out':
+      gain /= fan_out
+    elif mode == 'fan_avg':
+      gain /= (fan_in + fan_out) / 2
+    if distribution == 'truncated_normal':
+      # constant from scipy.stats.truncnorm.std(a=-2, b=2, loc=0., scale=1.)
+      stddev = backend.numpy.sqrt(gain) / .87962566103423978
+      return (backend.random.truncated_normal(rng, -2, 2, shape) * stddev).astype('float32')
+    elif distribution == 'normal':
+      return (backend.random.normal(rng, shape, dtype) * backend.numpy.sqrt(gain)).astype('float32')
```

Comment:
dtype is an udefined variable here

Reply:
Yeah, those were from a older version of the code before I followed the standard of the other initializer that cast everything to float32, again weird that no test caught it. Fixed